### PR TITLE
refactor(api-kit): convert to typescript

### PR DESCRIPTION
**Pull request by naorpeled**

### CODE REVIEW

File: packages/api-kit/package.json [hunk 1]
```diff
@@ -2,7 +2,10 @@
   "name": "@greenpress/api-kit",
   "version": "2.0.15",
   "description": "API-Kit package to help with greenpress infrastructure and reuse capabilities across services",
-  "main": "index.js",
+  "main": "dist/index.js",
+  "scripts": {
+    "build": "tsc"
```
Comment (davidmeirlevy) [inline packages/api-kit/package.json:7]: Try to add also a "prebuild" with the same command.

File: packages/api-kit/src/user-middlewares.ts [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+
+function verifyUser(req: RequestWithUser, res: Response, next: NextFunction): void {
+  if (req.user) {
+    next();
+  } else {
+    res.status(401).end();
+  }
+}
+
+module.exports = {
```
Comment (davidmeirlevy) [inline packages/api-kit/src/user-middlewares.ts:22]: Consider using just export for the actual functions and remove module.exports

File: packages/api-kit/src/router.ts [hunk 3]
```diff
@@ -0,0 +1,7 @@
+import { Router } from 'express'
+
+module.exports = {
```
Comment (davidmeirlevy) [inline packages/api-kit/src/router.ts:3]: Consider just export the function without module.exports.

**Review by davidmeirlevy — CHANGES_REQUESTED** [review]
Thanks!!

I believe my comment regarding the "prebuild" step will allow the tests to pass.

The other comments are only about using module.export, since if it's moving to TS we can replace it with regular export.

File: packages/api-kit/.npmignore [hunk 4]
```diff
@@ -0,0 +1,3 @@
+node_modules
+.idea
+.vscode
```
Comment (davidmeirlevy) [inline packages/api-kit/.npmignore:3]: Add the "dist" to npmignore.

**Review by davidmeirlevy — COMMENTED** [review]
For some reason, other apps cannot reach to "@greenpress/api-kit/router" even though it's a file at the dist folder.
We need to find a way to make the dist folder as the main folder so it can be used.
I believe it's possible but I'm far from a computer so I can't find the solution, but I believe that there's a valid way to solve it.

**Comment by naorpeled** [comment]
> For some reason, other apps cannot reach to "@greenpress/api-kit/router" even though it's a file at the dist folder. We need to find a way to make the dist folder as the main folder so it can be used. I believe it's possible but I'm far from a computer so I can't find the solution, but I believe that there's a valid way to solve it.

Hey,
I did some research regarding this issue, unfortunately after some trial and error I still haven't managed to resolve it.
Will continue looking into it when I have time.

**Comment by davidmeirlevy** [comment]
@naorpeled you can just search for the usage of this package inside the monorepo and import from `dist`.

**Comment by naorpeled** [comment]
> @naorpeled you can just search for the usage of this package inside the monorepo and import from `dist`.

I thought of that solution, but I don't really like breaking the usage.
I think I will do it as a quick fix and submit a separate PR once I find a way to resolve it.
What do you think?

**Comment by davidmeirlevy** [comment]
I think it's a good idea!
And also, I believe it's an important piece of knowledge regarding building node packages.

**Comment by naorpeled** [comment]
@davidmeirlevy I commited a fix that seems to pass the tests.
I'll clean the code and once I feel it's ready for review I'll let you know :)

File: packages/api-kit/.npmignore [hunk 5]
```diff
@@ -0,0 +1,4 @@
...
+dist
```
Comment (davidmeirlevy) [inline packages/api-kit/.npmignore:4]: You should remove dist from npmignore, because the dist folder should be published to the npm package.

**Review by davidmeirlevy — APPROVED** [review]

**Comment by davidmeirlevy** [comment]
Looks good. Have you tested it?
I mean, did you ran the app?

File: packages/api-kit/.npmignore (hunk 5 above)
Comment (naorpeled) [reply packages/api-kit/.npmignore]: Makes sense,
when you wrote me to add it there I wasn't sure if I was missing anything

**Comment by naorpeled** [comment]
> Looks good. Have you tested it? I mean, did you ran the app?

I've had issues with running `greenpress start dev`,
getting `Server took to long to run`.
Not sure why it's happening.

**Comment by davidmeirlevy** [comment]
@naorpeled if you have any questions regarding a local dev environment - ask.

**Comment by naorpeled** [comment]
@davidmeirlevy
Hey,
I managed to run the project properly (looked at some endpoints and saw that what I'm expecting is actually happening),
so I think you can merge the changes now :)

**Comment by davidmeirlevy** [comment]
merging!

Thanks!!

**Pull request merged**